auth/idtoken/idtoken.go

```diff
@@ -87,18 +87,18 @@ func (o *Options) validate() error {
 	return nil
 }
 
-// NewTokenProvider creates a [cloud.google.com/go/auth.TokenProvider] that
+// NewCredentials creates a [cloud.google.com/go/auth.Credentials] that
 // returns ID tokens configured by the opts provided. The parameter
 // opts.Audience may not be empty.
-func NewTokenProvider(opts *Options) (auth.TokenProvider, error) {
+func NewCredentials(opts *Options) (*auth.Credentials, error) {
 	if err := opts.validate(); err != nil {
 		return nil, err
 	}
 	if b := opts.jsonBytes(); b != nil {
-		return tokenProviderFromBytes(b, opts)
+		return credsFromBytes(b, opts)
 	}
 	if metadata.OnGCE() {
-		return computeTokenProvider(opts)
+		return computeCredentials(opts)
 	}
 	return nil, fmt.Errorf("idtoken: couldn't find any credentials")
 }
```

Comment:
I believe the previous PR in this series may have introduced a potential bug that I just noticed now: Search the package for

```
udp = auth.CredentialsPropertyFunc(creds.QuotaProjectID)
```

and you should see three usages where `udp` is assigned to `UniverseDomainProvider`. Maybe it should be assigned to `QuotaProjectIDProvider` instead?